src/core/chart.js

```diff
@@ -0,0 +1,252 @@
+      c3Data.axes[timeseriesName] = "y";
+    }
+    else if((!y2Units) || timeseries.units == y2Units) {
+      y2Units = timeseries.units;
+      c3Data.axes[timeseriesName] = "y2";
+    }
+    else {
+      throw new Error("Error: too many data axes required for graph");
+    }
+  }
+  
+  //whole-graph display options: axis formatting and tooltip behaviour
+  var c3Axis = {};
+  c3Axis.x = monthlyXAxis;
+  c3Axis.y = formatYAxis(yUnits);
+  if(y2Units) { 
+    c3Axis.y2 = formatYAxis(y2Units);
+    }
+    
+  var c3Tooltip = {format: {}};
+  c3Tooltip.grouped = "true";
+  c3Tooltip.format.value = tooltipDisplayNumbersWithUnitsFunction(c3Data.axes, c3Axis);
+  
+  return {
+    data: c3Data,
+    tooltip: c3Tooltip,
+    axis: c3Axis
+  }; 
+};
+
+/*
+ * Helper function for parseTimeseriesForC3.
+ * Accepts a dataseries object with 1, 4, or 12 timestamp:value pairs
+ * and returns an array with twelve values in order by timestamp,
+ * repeating values as necessary to get a monthly-resolution sequence.
+ */
+var getMonthlyData = function(data, timescale = "monthly") {
+
+  var expectedTimestamps = {"monthly": 12, "seasonal": 4, "yearly": 1};
+  var monthlyData = [];
+  var timestamps = Object.keys(data).sort();
+  
+  if(timestamps.length == 17) {
+    throw new Error("Error: concatenated 17-point chronology.");
+  }
+  
+  if(timestamps.length != expectedTimestamps[timescale]) {
+    throw new Error("Error: inconsistent time resolution in data");
+  }
+  
+  for(var i = 0; i < 12; i++) {
+    var mapped = Math.ceil((timestamps.length / 12.0) * (i + 1)) - 1;
+    monthlyData.push(data[timestamps[mapped]]);
+  }
+  
+  //Seasonal timeseries need one month of winter removed from the beginning of the
+  //year and added at the end, since winter wraps around the calendar new year.
+  if(timescale == "seasonal") {
+    monthlyData = monthlyData.slice(1, 12);
+    monthlyData.push(data[timestamps[0]]);
+  }
+  
+  return monthlyData;
+};
+
+/*
+ * Helper function for parseTimeseriesForC3. Given a set of timeserieses 
+ * to be graphed and metadata about each timeseries, returns a function 
+ * that generates the shortest name necessary to distinguish a particular
+ * timeseries from all others being shown on the same chart.
+ * 
+ * For example, when graphing monthly, seasonal, and yearly means for 
+ * otherwise identical data run, only "monthly", "seasonal," and "yearly"
+ * need to appear in the graph legend. But if graphing multiple variables,
+ * the graph legend will need to display variable names as well.
+ * 
+ * Timeseries names include any descriptive  metadata that vary between 
+ * timeseries and leave out any metadata that doesn't. They end with "mean".
+ */
+//TODO: special case climatological period to display as (XXXX-XXXX)
+//TODO: possibly cue descriptors to appear in a specific order?
+// "Tasmin Monthly Mean" sounds better than "Monthly Tasmin Mean".
+var shortestUniqueTimeseriesNamingFunction = function (metadata, data) {
```

Comment:
This is an interesting function, and I think that it's a really nice way to separate out some funky heuristics that will *really* improve usability and readability of the charts. It *does* have a lot of weird cases though, so if you don't have tests in place yet, I'd advise that you write some.